File: C/2020/12限/file2.c
//  file2.c
//  cl  file2.c
#include <stdio.h>
#define STRBUF 512

int main() {
    FILE *fp;
    fp = fopen("file.c", "r");

    // ファイルの読み込み
    char buf[STRBUF];
    char *r;
    //
    /*
    r = fgets(buf, STRBUF, fp);
    printf("%p : %s \n", r, buf);
    */
    //while((r = fgets(buf, STRBUF, fp)) != NULL) {
    while(fgets(buf, STRBUF, fp) != NULL) {
        printf(": %s ", buf);
    }

    //
    fclose(fp);

    return 0;
}

File: PhpBeginner/2021/EarlyPeriod(Tue.)/file_2.php
<?php   // file_2.php
//
//$fp = fopen('dummy', 'r');
$fp = fopen(__FILE__, 'r');
var_dump($fp);

//$line = fgets($fp);
//var_dump($line);

while(true) {
    $line = fgets($fp);
    if (true === feof($fp)) {
        break;
    }
    var_dump($line);
}

/*
while(false !== ($line = fgets($fp))) {
    var_dump($line);
}
*/

/*
while(true) {
    $line = fgets($fp);
    if (false === $line) {
        break;
    }
    var_dump($line);
}
*/

$r = fclose($fp);
